google/cloud/forseti/services/scanner/dao.py

```diff
@@ -123,41 +238,55 @@ def create(self, violations, inventory_index_id):
                         violation.get('violation_data', ''),
                     )
 
-                    violation = self.TBL_VIOLATIONS(
+                    violation = Violation(
+                        created_at_datetime=created_at_datetime,
+                        full_name=violation.get('full_name'),
                         inventory_index_id=inventory_index_id,
+                        resource_data=violation.get('resource_data'),
                         resource_id=violation.get('resource_id'),
                         resource_type=violation.get('resource_type'),
-                        full_name=violation.get('full_name'),
-                        rule_name=violation.get('rule_name'),
                         rule_index=violation.get('rule_index'),
-                        violation_type=violation.get('violation_type'),
+                        rule_name=violation.get('rule_name'),
+                        scanner_index_id=scanner_index_id,
                         violation_data=json.dumps(
                             violation.get('violation_data')),
-                        resource_data=violation.get('resource_data'),
                         violation_hash=violation_hash,
-                        created_at_datetime=created_at_datetime
+                        violation_type=violation.get('violation_type')
                     )
 
                     session.add(violation)
 
-        def list(self, inventory_index_id=None):
+        def list(self, inventory_index_id=None, scanner_index_id=None):
             """List all violations from the db table.
 
             Args:
                 inventory_index_id (str): Id of the inventory index.
+                scanner_index_id (str): Id of the scanner index.
 
             Returns:
                 list: List of Violation row entry objects.
             """
             with self.violationmaker() as session:
-                if inventory_index_id:
-                    return (
-                        session.query(self.TBL_VIOLATIONS).filter(
-                            self.TBL_VIOLATIONS.inventory_index_id ==
-                            inventory_index_id).all()
-                    )
-                return (
-                    session.query(self.TBL_VIOLATIONS).all())
+                query = None
+                if inventory_index_id and scanner_index_id:
```

Comment:
All the if/elif/else code here can be simplified if you pull this into a helper method.  `build_list_query`
```
if scanner_index_id:
    return foo_query

if not (inventory_index_id or scanner_index_id):
    return bar_query

return final_possible_query
```